backup/backup_test.go

```diff
@@ -12,94 +14,175 @@ import (
+func (o *getUUIDSetMockOp) PrepareRestore(_ context.Context) error {
 	panic("not implemented")
 }
 
-func (o *choosePodMockOp) LoadDump(ctx context.Context, dir string) error {
+func (o *getUUIDSetMockOp) LoadDump(ctx context.Context, dir string) error {
 	panic("not implemented")
 }
 
-func (o *choosePodMockOp) LoadBinlog(ctx context.Context, binlogDir, tmpDir string, restorePoint time.Time) error {
+func (o *getUUIDSetMockOp) LoadBinlog(ctx context.Context, binlogDir, tmpDir string, restorePoint time.Time) error {
 	panic("not implemented")
 }
 
-func (o *choosePodMockOp) FinishRestore(_ context.Context) error {
+func (o *getUUIDSetMockOp) FinishRestore(_ context.Context) error {
 	panic("not implemented")
 }
 
-func TestChoosePod(t *testing.T) {
-	makePod := func(ready bool) *corev1.Pod {
-		pod := &corev1.Pod{}
-		if !ready {
-			return pod
-		}
-		pod.Status.Conditions = []corev1.PodCondition{{
-			Type:   corev1.PodReady,
-			Status: corev1.ConditionTrue,
-		}}
+func makePod(ready bool) *corev1.Pod {
+	pod := &corev1.Pod{}
+	if !ready {
 		return pod
 	}
+	pod.Status.Conditions = []corev1.PodCondition{{
+		Type:   corev1.PodReady,
+		Status: corev1.ConditionTrue,
+	}}
+	return pod
+}
 
-	makePod1 := func(ready bool) []*corev1.Pod {
-		return []*corev1.Pod{makePod(ready)}
-	}
-	makePod3 := func(ready0, ready1, ready2 bool) []*corev1.Pod {
-		return []*corev1.Pod{makePod(ready0), makePod(ready1), makePod(ready2)}
-	}
+func makePod1(ready bool) []*corev1.Pod {
+	return []*corev1.Pod{makePod(ready)}
+}
 
-	var lastOp *choosePodMockOp
+func makePod3(ready0, ready1, ready2 bool) []*corev1.Pod {
+	return []*corev1.Pod{makePod(ready0), makePod(ready1), makePod(ready2)}
+}
+
+func TestGetUUIDSet(t *testing.T) {
+	var lastOp *getUUIDSetMockOp
 	newOperator = func(host string, port int, user, password string, threads int) (bkop.Operator, error) {
-		lastOp = &choosePodMockOp{uuid: "123"}
+		lastOp = &getUUIDSetMockOp{uuid: "123"}
 		return lastOp, nil
 	}
 
-	makeBM := func(replicas, current int, bkup mocov1beta2.BackupStatus) *BackupManager {
+	testCases := []struct {
+		name string
+		pods []*corev1.Pod
+
+		expected map[string]string
+	}{
+		{"empty", []*corev1.Pod{}, map[string]string{}},
+		{"single-not-ready", makePod1(false), map[string]string{}},
+		{"single-ready", makePod1(true), map[string]string{"0": "123"}},
+		{"triple-not-ready", makePod3(false, false, false), map[string]string{}},
+		{"triple-1st-ready", makePod3(true, false, false), map[string]string{"0": "123"}},
+		{"triple-2nd-ready", makePod3(false, true, false), map[string]string{"1": "123"}},
+		{"triple-3rd-ready", makePod3(false, false, true), map[string]string{"2": "123"}},
+		{"triple-all-ready", makePod3(true, true, true), map[string]string{"0": "123", "1": "123", "2": "123"}},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			lastOp = nil
+			bm := &BackupManager{
+				cluster: &mocov1beta2.MySQLCluster{},
+				status: bkop.ServerStatus{
```

Comment:
This status will be not used.

Reply:
The UUID field is used in bm.GetUUUIDSet()